db/attachment.go

```diff
@@ -125,13 +129,18 @@ func (db *Database) storeAttachments(doc *document, body Body, generation int, p
 // Attempts to retrieve ancestor attachments for a document.  First attempts to find and use a non-pruned ancestor.
 // If no non-pruned ancestor is available, checks whether the currently active doc has a common ancestor with the new revision.
 // If it does, can use the attachments on the active revision with revpos earlier than that common ancestor.
-func (db *Database) retrieveAncestorAttachments(doc *document, parentRev string, docHistory []string) map[string]interface{} {
+func (db *Database) retrieveAncestorAttachments(doc *document, parentRev string, docHistory []string) (map[string]interface{}, error) {
 
 	var parentAttachments map[string]interface{}
 	// Attempt to find a non-pruned parent or ancestor
-	parent, _ := db.getAvailableRev(doc, parentRev)
+	parent, revid, _ := db.getAvailableRev(doc, parentRev)
 	if parent != nil {
-		parentAttachments = GetBodyAttachments(parent)
+
+		doc, err := db.revisionCache.Get(doc.ID, revid)
```

Comment:
getAvailableRev should have already populated the _attachments property in parent, so an additional revCache fetch shouldn't be needed here.